lib/utilities/lint-fix.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const execa = require('execa');
+
+function eslint(filePaths) {
+  return execa('eslint', [...filePaths, '--fix'], { cwd: process.cwd(), preferLocal: true });
+}
+
+function emberTemplateLint(filePaths) {
+  return execa('ember-template-lint', [...filePaths, '--fix'], { cwd: process.cwd(), preferLocal: true });
+}
```

Comment:
I was thinking that we should check for `lint:fix` and run it (that will support more things than just eslint/ember-template-lint if they've configured them)

Reply:
I considered the same thing however there were a couple of things that steered me away from this approach:

- It could affect files beyond what's generated. I wouldn't expect `ember generate` to lint fix unrelated files. 
- The performance difference. I expect that linting the whole app is likely to be slower than targeting only what's changed.

Though you make a good point about `lint:fix` providing support for more than just the default lint tools. Would `lint:fix` be your preference?

Reply:
Thanks for the feedback, and for reviewing it so quickly 👍

Reply:
> Would lint:fix be your preference?

Yes.

RE: the two main concerns:

> It could affect files beyond what's generated. I wouldn't expect ember generate to lint fix unrelated files.

IMHO we have to think of these as idempotent. It _shouldn't_ change unrelated files because you _shouldn't_ be in a "failing linting" state.

> The performance difference. I expect that linting the whole app is likely to be slower than targeting only what's changed.

Ya, this is definitely true though I think the usage of `--cache` by default in `lint:js` should mitigate.

Reply:
Could we combine the two? Pseudo code `execa('npm', ['run', 'lint:fix', '--', ...filePaths])`. I'm not sure how the npm-run-all package passes args.

Reply:
I looked into this however as `lint:fix` calls the `lint:*:fix` scripts and they already use a `.`, e.g. `eslint . --fix`, I think that will always guarantee that everything will be lint fixed.

I tested this by running `./node_modules/.bin/eslint . --fix ./app/router.js` and the additional path had no effect.